enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/tracing/exporters/OTLPExporter.java

```diff
@@ -0,0 +1,107 @@
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.choreo.connect.enforcer.config.ConfigHolder;
+import org.wso2.choreo.connect.enforcer.tracing.RateLimitingSampler;
+import org.wso2.choreo.connect.enforcer.tracing.TracerBuilder;
+import org.wso2.choreo.connect.enforcer.tracing.TracingConstants;
+import org.wso2.choreo.connect.enforcer.tracing.TracingException;
+
+import java.net.URISyntaxException;
+import java.util.Map;
+
+/**
+ * Trace Exporter for Open Telemetry Protocol
+ */
+public class OTLPExporter implements TracerBuilder {
+
+    private static final Logger LOGGER = LogManager.getLogger(OTLPExporter.class);
+
+    private static OTLPExporter exporter;
+
+    public static OTLPExporter getInstance() {
+        if (exporter == null) {
+            synchronized (new Object()) {
+                if (exporter == null) {
+                    exporter = new OTLPExporter();
+                }
+            }
+        }
+        return exporter;
+    }
+
+    @Override
+    public OpenTelemetrySdk initSdk(Map<String, String> properties) throws TracingException {
+        String host = properties.get(TracingConstants.CONF_HOST);
+        String authHeaderName = properties.get(TracingConstants.CONF_AUTH_HEADER_NAME);
+        String authHeaderValue = properties.get(TracingConstants.CONF_AUTH_HEADER_VALUE);
+        String connectionString = properties.get(TracingConstants.CONF_CONNECTION_STRING);
+        ConfigHolder conf = ConfigHolder.getInstance();
+        OtlpGrpcSpanExporterBuilder otlpGrpcSpanExporterBuilder = OtlpGrpcSpanExporter.builder();
+
+        if (connectionString != null) {
+            otlpGrpcSpanExporterBuilder.setEndpoint(connectionString);
+        } else {
+            try {
+                int port = Integer.parseInt(properties.get(TracingConstants.CONF_PORT));
+                String endpoint = new URIBuilder().setHost(host).setPort(port)
+                        .setScheme(host.contains("https") ? "https" : "http")
+                        .build().toString();
+                otlpGrpcSpanExporterBuilder.setEndpoint(endpoint);
+            } catch (URISyntaxException | NumberFormatException e) {
+                throw new TracingException("Couldn't initialize the OTLP exporter. Invalid endpoint definition", e);
+            }
+        }
+
+        if (authHeaderName != null && authHeaderValue != null) {
```

Comment:
Don't we have to check authHeaderName isblank too?

Reply:
@AmaliMatharaarachchi isblank means empty string ? 

If no `authHeaderName` and `authHeaderValue` fields are added to config, we'll get `null` for those 2 variables. If the users add those to config but without a value, then we get empty strings. Do we need to handle that too ? Something like the below config.

```toml
[tracing]
  # Enable/Disable tracing in Choreo Connect
  enabled = true
  # Type of tracer exporter (e.g: azure, zipkin). Use zipkin type for Jaeger as well.
  type = "otlp"
  # configurations for zipkin tracer type
  [tracing.configProperties]
    # maximum length of the request path to extract and include in the HttpUrl tag.
    maxPathLength = "256"
    # remote url for publishing traces
    connectionString = "https://otlp.nr-data.net"
    # auth header name
    authHeaderName = ""
    # auth header value
    authHeaderValue = ""
    # library Name to be tagged in traces (`otel.library.name`).
    instrumentationName = "CHOREO-CONNECT"
    # Maximum number of sampled traces per second string
    maximumTracesPerSecond = "2"
```

Reply:
just thought it would be better to validate the empty string. Btw, didn't we add an example to the default_config.template?

Reply:
@AmaliMatharaarachchi added additional check for empty string and added an example to config.toml.template.